crates/forge_services/src/agent_registry.rs

```diff
@@ -1,71 +1,36 @@
 use std::sync::Arc;
 
-use dashmap::DashMap;
 use forge_app::AgentRepository;
 use forge_app::domain::AgentId;
 use forge_domain::{Agent, AppConfigRepository, ProviderRepository};
 use tokio::sync::RwLock;
 
-/// AgentRegistryService manages the active-agent ID and a registry of runtime
-/// Agents in-memory. It lazily loads agents from AgentRepository on first
-/// access.
+/// AgentRegistryService manages the active-agent ID and coordinates with
+/// AgentRepository to provide runtime Agent instances.
+///
+/// The service converts AgentDefinitions from the repository into runtime
+/// Agents by applying default provider and model configurations.
+/// Caching is handled internally by the AgentRepository.
 pub struct ForgeAgentRegistryService<R> {
-    // Infrastructure dependency for loading agent definitions
+    // Infrastructure dependency for loading agent definitions and config
     repository: Arc<R>,
 
-    // In-memory storage for agents keyed by AgentId string
-    // Lazily initialized on first access
-    // Wrapped in RwLock to allow invalidation
-    agents: RwLock<Option<DashMap<String, Agent>>>,
```

Comment:
AgentDefinition to AgentConversation require IO operations hence we should cache. Cache only where we know there are going to IO operations.